### DISCUSSION TITLE
Binding return from REST call to session variable or form field

### DISCUSSION THREAD

**Discussion by offpiste-reese** [Budibase/budibase#5405] [Help] [answered]
I am making a  REST  call that returns the following JSON
{
  "sessionId": "0YK~A4_zGjSbf~eMlV2hKvhlRcbfV3Dg"
}

What is the syntax to store this value in a Session variable or bind it to a form field?  I have tried several variations of the following. 
![image](https://user-images.githubusercontent.com/103227709/163241221-ecad4243-8cd0-4ad5-98c6-e59bc323460c.png)

Additionally, is there a way to populate a form field with the result without changing pages?

**Comment by melohagan** [comment, marked as answer]
Hey @offpiste-reese 

Remove the square brackets and the syntax should be correct: `{{ Action 1.Query result.sessionId }}`

A feature is currently in the works to support on change: https://github.com/Budibase/budibase/issues/5392

In the meantime, a potential workaround would be to set the default value of your form field to `{{ State.sessionId }}`
and then add an additional action **Clear Form** after the Update State.
This will trigger a re-render of that form, and pull in the recently updated state into the field.

**Reply by offpiste-reese** [reply]
I tried the syntax you suggested prior to posting, it did not work.  However this syntax did:  {{ Action 1.Query result .data.0.[sessionId]}}

I figured out that if I hid the text field and then showed it when there was a value in my session variable, it would display when I updated the session variable.  That's an ok way to do it. 

Thanks.

**Reply by melohagan** [reply]
@offpiste-reese Apologies, I forgot myself about the data field. It is stated in the docs here: https://docs.budibase.com/docs/referencing
